plugins/scaffolder-backend/src/scaffolder/actions/builtin/filesystem/delete.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createTemplateAction } from '../../createTemplateAction';
+import { InputError } from '@backstage/errors';
+import fs from 'fs-extra';
+
+export const createFilesystemDeleteAction = () => {
+  return createTemplateAction<{ files: string[] }>({
+    id: 'fs:delete',
+    schema: {
+      input: {
+        required: ['files'],
+        type: 'object',
+        properties: {
+          files: {
+            title: 'Files',
+            description: 'A list of files or directories that will be deleted',
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      if (!Array.isArray(ctx.input?.files)) {
+        throw new InputError('files must be an Array');
+      }
+
+      for (const file of ctx.input.files) {
+        if (file.includes('..')) {
```

Comment:
I wonder if we can check if the file path is absolute too and also throw an error because it's outside of workspace.

Reply:
I didn't add specific checks on absolute path because the prefix is fixed, so it always start with workspace, so you could only go back from there.

```const filepath = `${ctx.workspacePath}/${file}`;```

we could run some checks on the outcome of `filepath` maybe

Reply:
yeah we've added some special logic for this in the techdocs part to avoid security issues. @OrkoHunter @iamEAP can you remember where the code is that I'm talking about 😛

Reply:
check this pr https://github.com/backstage/backstage/pull/6204

Reply:
thanks @freben, that util makes lot of sense 🚀

@benjaminr I've updated the PR with this, removing my custom check on `contains ".."` :) 🙏